From 146e4859e6a95c1929889e4b451b9fdc1c161642 Mon Sep 17 00:00:00 2001
From: Jeff Moyer <jmoyer@redhat.com>
Date: Thu, 28 Aug 2014 19:16:07 -0400
Subject: [virt] virtio_blk, blk-mq: remove alloc_hctx and free_hctx methods

Message-id: <1409253371-23161-18-git-send-email-jmoyer@redhat.com>
Patchwork-id: 89860
O-Subject: [RHEL7 PATCH 17/21] virtio-blk, blk-mq: remove alloc_hctx and free_hctx methods
Bugzilla: 1105698
RH-Acked-by: Fam Zheng <famz@redhat.com>

This is a backport of the virtio-blk portion of the following commit.
This is related to bug 1105698.

  commit cdef54dd85ad66e77262ea57796a3e81683dd5d6
  Author: Christoph Hellwig <hch@lst.de>
  Date:   Wed May 28 18:11:06 2014 +0200

    blk-mq: remove alloc_hctx and free_hctx methods

    There is no need for drivers to control hardware context allocation
    now that we do the context to node mapping in common code.

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/virtio_blk.c b/drivers/block/virtio_blk.c
index 9f340fa..c8f286e 100644
--- a/drivers/block/virtio_blk.c
+++ b/drivers/block/virtio_blk.c
@@ -497,8 +497,6 @@ static int virtblk_init_request(void *data, struct request *rq,
 static struct blk_mq_ops virtio_mq_ops = {
  .queue_rq = virtio_queue_rq,
  .map_queue = blk_mq_map_queue,
- .alloc_hctx = blk_mq_alloc_single_hw_queue,
- .free_hctx = blk_mq_free_single_hw_queue,
  .complete = virtblk_request_done,
  .init_request = virtblk_init_request,
 };
-- 
1.7.1